invenio_rdm_records/services/pids/providers/base.py

```diff
@@ -152,17 +187,36 @@ def get_status(self, identifier, **kwargs):
         """Get the status of the identifier."""
         return self.get(identifier, **kwargs).status
 
-    def validate(self, identifier=None, provider=None, client=None, **kwargs):
+    def validate(
+        self, record, identifier=None, provider=None, client=None, **kwargs
+    ):
         """Validate the attributes of the identifier.
 
         :returns: A tuple (success, errors). The first specifies if the
                   validation was passed successfully. The second one is an
                   array of error messages.
         """
         if provider and provider != self.name:
-            errors = [
-                _(f"Provider name {provider} does not match {self.name}")
-            ]
-            return False, errors
+            logging.error(f"Configuration error: provider name {provider} "
+                          f"does not match {self.name}.")
+            raise   # configuration error
```

Comment:
is a config error but it will raise at run time when validating a record pids... should we raise with some message?

Reply:
Here what is happening is that if we raise with a specific message, it might be returned in the API response.
For `400`, it is indeed what we want.

For `500`, what you want to have is to log the error to find it, but to return simply 500 Internal Server error. You don't want, normally, to give specific info on the error, it might a vector of attack e.g. in case of misconfiguration. At least what I personally think :)